app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/ApproveQuotesUseCase.kt

```diff
@@ -0,0 +1,50 @@
+package com.hedvig.app.feature.offer.ui.checkout
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.SignQuotesMutation
+import com.hedvig.app.feature.offer.OfferRepository
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+import java.time.LocalDate
+
+class ApproveQuotesUseCase(
+    private val apolloClient: ApolloClient,
+    private val offerRepository: OfferRepository
+) {
+
+    sealed class ApproveQuotesResult {
+        data class Success(val date: LocalDate?) : ApproveQuotesResult()
+        sealed class Error : ApproveQuotesResult() {
+            data class GeneralError(val message: String?) : Error()
```

Comment:
How about we add a default argument here? Then we don't need to have the form `GeneralError(null)` later 😄 

```suggestion
            data class GeneralError(val message: String? = null) : Error()
```